domain/config.go

```diff
@@ -60,16 +61,139 @@ type Config struct {
+			PreferredPoolIDs:                 []uint64{},
+			MaxPoolsPerRoute:                 4,
+			MaxRoutes:                        20,
+			MaxSplitRoutes:                   3,
+			MinPoolLiquidityCap:              0,
+			RouteCacheEnabled:                true,
+			CandidateRouteCacheExpirySeconds: 1200,
+			RankedRouteCacheExpirySeconds:    45,
+			DynamicMinLiquidityCapFiltersDesc: []DynamicMinLiquidityCapFilterEntry{
+				{
+					MinTokensCap: 1000000,
+					FilterValue:  75000,
+				},
+				{
+					MinTokensCap: 250000,
+					FilterValue:  15000,
+				},
+				{
+					MinTokensCap: 10000,
+					FilterValue:  1000,
+				},
+				{
+					MinTokensCap: 1000,
+					FilterValue:  10,
+				},
+				{
+					MinTokensCap: 1,
+					FilterValue:  1,
+				},
+			},
+		},
+		Pricing: &PricingConfig{
+			CacheExpiryMs:             2000,
+			DefaultSource:             0,
+			DefaultQuoteHumanDenom:    "usdc",
+			MaxPoolsPerRoute:          4,
+			MaxRoutes:                 3,
+			MinPoolLiquidityCap:       1000,
+			CoingeckoUrl:              "https://prices.osmosis.zone/api/v3/simple/price",
+			CoingeckoQuoteCurrency:    "usd",
+			WorkerMinPoolLiquidityCap: 1,
+		},
+		Passthrough: &passthroughdomain.PassthroughConfig{
+			NumiaURL:                     "https://public-osmosis-api.numia.dev",
+			TimeseriesURL:                "https://stage-proxy-data-api.osmosis-labs.workers.dev",
+			APRFetchIntervalMinutes:      5,
+			PoolFeesFetchIntervalMinutes: 5,
+		},
+		GRPCIngester: &GRPCIngesterConfig{
+			Enabled:                        true,
+			MaxReceiveMsgSizeBytes:         16777216,
+			ServerAddress:                  ":50051",
+			ServerConnectionTimeoutSeconds: 10,
+			Plugins: []Plugin{
+				&OrderBookPluginConfig{
+					Enabled: false,
+					Name:    orderbookplugindomain.OrderBookPluginName,
+				},
+			},
+		},
+		OTEL: &OTELConfig{
+			Enabled:     true,
+			Environment: "sqs-dev",
+		},
+		CORS: &CORSConfig{
+			AllowedHeaders: "Origin, Accept, Content-Type, X-Requested-With, X-Server-Time, Origin, Accept, Content-Type, X-Requested-With, X-Server-Time, Accept-Encoding, sentry-trace, baggage",
+			AllowedMethods: "HEAD, GET, POST, HEAD, GET, POST, DELETE, OPTIONS, PATCH, PUT",
+			AllowedOrigin:  "*",
+		},
+	}
+)
+
+// UnmarshalConfig handles the custom unmarshaling for the Config struct.
+// Additionally, it sets up environment variable mappings using reflection.
+// It also handles the Plugins field by decoding it using a custom decode hook.
+// It uses Viper to handle environment variables and reflection to automatically generate environment variable mappings.
+// CONTRACT: viper.ReadInConfig() is called before this function.
 func UnmarshalConfig() (*Config, error) {
-	var config Config
+	config := DefaultConfig
+
+	viper.SetEnvPrefix(envPrefix)
+	viper.AutomaticEnv()
+	viper.SetEnvKeyReplacer(strings.NewReplacer("-", "_", ".", "_"))
+
+	// Set up environment variable mappings using reflection
+	bindEnvRecursive(reflect.ValueOf(&config), "")
```

Comment:
Note to reviewer: this logic handles dynamically creating environment variable bindings to to allow overriding any config value from the default config